tools/serverpod_cli/lib/src/analyzer/models/definitions.dart

```diff
@@ -89,16 +89,32 @@ class ClassDefinition extends SerializableModelDefinition {
         .firstWhere((element) => element.name == name, orElse: () => null);
   }
 
+  /// Returns the `ClassDefinition` of the parent class.
+  /// If there is no parent class, `null` is returned.
   ClassDefinition? get parentClass {
     var extendsClass = this.extendsClass;
     if (extendsClass is! ResolvedInheritanceDefinition) return null;
 
     return extendsClass.classDefinition;
   }
 
+  /// Returns a list of all fields in the parent class.
+  /// If there is no parent class, an empty list is returned.
   List<SerializableModelFieldDefinition> get parentFields =>
       parentClass?.fields ?? [];
 
+  /// Returns a list of all fields in this class, including inherited fields.
+  /// It ensures that the 'id' field, if present, is always included at the beginning of the list.
+  List<SerializableModelFieldDefinition> get allFields {
```

Comment:
Discussion/Question: The `fields` getter and this getter are a bit confusing now. Should we rename the `fields` getter to better reflect that this is only includes fields that are defined in the class?

Reply:
or we change `allFields` to be more descriptive?